### PR TITLE
BHBC-1883: secure species under the persecution or harm rules

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

- https://quartech.atlassian.net/browse/BHBC-1883

## Description of relevant changes

- Mechanism is implemented to secure spatial records based on trasnformation rules specific to their species

### Tests

- [/] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [/] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [/] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [/] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

Use this sample zip file for a test sample with combination of species that are protected and unprotected

To test:
* upload the zip file below through postman (intake endpoint)
* Look at the database

[DwCA_with_EML_MultipleSpecies.zip](https://github.com/bcgov/biohubbc-platform/files/9304161/DwCA_with_EML_MultipleSpecies.zip)

### CODE REVIEW

File: api/src/repositories/spatial-repository.ts [hunk 1]
```diff
@@ -148,6 +186,44 @@ export class SpatialRepository extends BaseRepository {
...
+      `;
+
+    const response = await this.connection.sql<{ security_transform_submission_id: number }>(sqlStatement);
+
+    if (response.rowCount !== 1) {
+      throw new ApiExecuteSQLError(
+        'Failed to insert security transform submission id and submission spatial component id',
+        [
+          'SpatialRepository->insertSecurityTransformSubmissionRecord',
+          'rowCount was null or undefined, expected rowCount >= 1'
```
Comment (NickPhura) [inline api/src/repositories/spatial-repository.ts]: nit: expected rowCount = 1

File: api/src/repositories/spatial-repository.ts [hunk 2]
```diff
@@ -169,6 +245,27 @@ export class SpatialRepository extends BaseRepository {
...
+   * @return {*}  {Promise<ITransformRow[]>}
+   * @memberof SpatialRepository
+   */
+  async runSecurityTransformOnSubmissionId(submissionId: number, transform: string): Promise<ITransformSecureRow[]> {
+    const response = await this.connection.query(transform, [submissionId]);
+
+    if (response.rowCount <= 0) {
+      throw new ApiExecuteSQLError('Failed to run security transform on submission id', [
+        'SpatialRepository->runSecurityTransformOnSubmissionId',
+        'rowCount was null or undefined, expected rowCount >= 1'
```
Comment (NickPhura) [inline api/src/repositories/spatial-repository.ts:262]: nit: expected rowCount > 1

**Comment by NickPhura** [comment]
A few minor things we can tweak in the dwc-service#create tests, but easier to chat about.

**Review by NickPhura — APPROVED** [review]

**Review by curtisupshall — APPROVED** [review]
Looks good! 🦌

**Pull request merged**